libcalico-go/lib/backend/k8s/resources/kubecontrollersconfig.go

```diff
@@ -30,7 +43,7 @@ const (
 )
 
 func NewKubeControllersConfigClient(c *kubernetes.Clientset, r *rest.RESTClient) K8sResourceClient {
-	return &customK8sResourceClient{
+	return &kubeControllersConfigClient{
```

Comment:
@huiyizzz rather than reimplement all of `customK8sResourceClient`, would it be possible to add a new option field to it?

e.g.,

```go
type customK8sResourceClient struct {
  validator Validator
}

// A Validator is a function that validates a resource and returns an error if it is not valid.
type Validator func(kvp KVPair) error
```

Then the Create / Update methods can have something like this in it:

```go
if c.Validator != nil {
  if err = c.Validator(kvp); err != nil {
    return nil, err
  }
}
```

WDYT?

Reply:
+1 (I was just writing a comment on how we should try to avoid "copying" `customK8sResourceClient`, @caseydavenport's solution is better than what I had)

Reply:
Oh right! Sounds like a better solution!